gitops/argocd/README.md

```diff
@@ -0,0 +1,43 @@
+# Argo CD registration
+
+This directory contains the logic used for registering the clusters to Argo CD.
+
+## Run
+
+The registration is thought to be triggered from [Pipelines as Code](https://pipelinesascode.com/) and a [Tekton PipelineRun](../pac/.tekton/argo-registration.yaml) is provided for the purpose in the .tekton directory.
+
+Alternatively the registration can be performed by manually calling the registration script in the image directory:
+
+```console
+ARGO_URL="https://argoserver.com" ARGO_USER="user" ARGO_PWD="xxxxxxxxx" DATA_DIR="/workspace" ./register.sh
+```
+
+DATA_DIR should point to a directory with a fork of this repository including
+- the kubeconfig files of the clusters to register in `gitops/credentials/kubeconfig/plnsvc`
+- two kustomization files for each cluster under two folders:
+  - `gitops/environment/plnsvc/$cluster/namespaces/kustomization.yaml` using `gitops/environment/plnsvc/base/namespaces` as base and any desired customization.
+  - `gitops/environment/plnsvc/$cluster/argocd-rbac/kustomization.yaml` using `gitops/environment/plnsvc/base/argocd-rbac` as base and any desired customization.
+
+~~~
+cat <<EOF > kustomization.yaml
+resources:
+- ../../base/namespaces
+EOF
+~~~
+
+~~~
+cat <<EOF > kustomization.yaml
+resources:
+- ../../base/argocd-rbac
+EOF
+~~~
+
+## Authentication
+
+Tekton PipelineRun relies on a secret named argocd-credentials in the same namespace as the PipelineRun for the authentication against the Argo CD server. This can be created as follows:
+~~~
+kubectl create secret generic argocd-credentials \
```

Comment:
How about switching to storing all credentials in `credentials/kubeconfig`, but encrypted?

Reply:
I still would not put them in a public repository. We may look at integrating with [Vault](https://www.vaultproject.io/docs/platform/k8s) moving forward but that is out of scope of  this PR